job/scriptsGen/normalizeSteps.js

```diff
@@ -86,50 +88,72 @@ function _normalizeNewFormatSteps(steps, defaultRuntime, onSuccess,
 
       var task = step.TASK;
 
+      // Normalize a string script into an array.
       if (_.isString(task.script))
         task.script = [task.script];
 
       task.runtime = task.runtime || {};
+
+      // Use the default task runtime if it is not specified.
       if (_.isUndefined(task.runtime.container))
         task.runtime.container = defaultIsContainer;
 
       if (task.runtime.container) {
-        if (_.isEmpty(task.runtime.options))
```

Comment:
This was the cause of the bug. We would copy all the options the first time and then append the same options again later.